src/providers/ApiProvider.ts

```diff
@@ -0,0 +1,26 @@
+import ApiService from '../services/ApiService';
+import { ApiType, Http, SearchData, TitleData } from '../interfaces';
+
+class ApiProvider {
+    search(q: string): Promise<SearchData> {
```

Comment:
in this case I don't know what does the "q" means, variable names should be meaningfull